moshi/moshi/modules/streaming.py

```diff
@@ -64,25 +64,26 @@ def is_streaming(self):
         return self._streaming_state is not None
 
     def set_streaming_propagate(self, streaming_propagate: bool):
+        """If set to False, this module will not switch to streaming if a parent module
+        is set to streaming mode, only if directly set to streaming mode."""
         self._streaming_propagate = streaming_propagate
 
     def _apply_named_streaming(self, fn: tp.Any):
-        def _handle_module(prefix: str, module: nn.Module, recurse: bool = True):
-            propagate = True
+        def _handle_module(prefix: str, module: nn.Module):
             if isinstance(module, StreamingModule):
-                if module._streaming_propagate:
-                    fn(prefix, module)
+                # If prefix is empty, we are the direct receiver of the streaming request,
+                # otherwise, we are inheriting from a parent and will stop if _streaming_propagate is False.
+                if not module._streaming_propagate and prefix != "":
```

Comment:
I find the semantics pretty confusing here: if A has propagate set, and has a children B for which propagate is not set, then calling `A._apply_named_streaming` results in `fn(A)` but not in `fn(B)`. I would have expected `fn(B)` to be called (but not `fn(C)` for any child C of B as B doesn't propagate down).
Anyway what is the use case for not propagating? Shouldn't we just always propagate?

Reply:
maybe a better name could be `streaming_detach`, closer to detaching a tensor from a compute graph for the backward. It means this entire module should not be consider a part of the main streaming path, in our case because the depformer is not streaming along the time dim, but some other dim

Reply:
+1 for `streaming_detach`

Reply:
done